chain/bitcoind_events.go

```diff
@@ -319,3 +357,306 @@ func (b *bitcoindZMQEvents) Stop() error {
+
+				height++
+			}
+
+		case <-b.quit:
+			return
+		}
+	}
+}
+
+// txEventHandlerRPC is a goroutine that uses the RPC client to check the
+// mempool for new transactions.
+func (b *bitcoindRPCPollingEvents) txEventHandlerRPC() {
+	defer b.wg.Done()
+
+	log.Info("Started polling for new bitcoind transactions via RPC.")
+	ticker := time.NewTicker(b.cfg.TxPollingInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			// After each ticker interval, we poll the mempool to
+			// check for transactions we haven't seen yet.
+			txs, err := b.client.GetRawMempool()
+			if err != nil {
+				log.Errorf("Unable to retrieve mempool txs: "+
+					"%v", err)
+				continue
+			}
+
+			// We'll scan through the most recent txs in the
+			// mempool to see whether there are new txs that we
+			// need to send to the client.
+			for _, txHash := range txs {
+				// If the transaction isn't in the local
+				// mempool, we'll send it to the client.
+				if b.mempool.contains(txHash) {
+					continue
+				}
+
+				// Grab full mempool transaction from hash.
+				tx, err := b.client.GetRawTransaction(txHash)
+				if err != nil {
+					log.Errorf("unable to fetch "+
+						"transaction %s from "+
+						"mempool: %v", txHash, err)
+					continue
+				}
+
+				// Add the transaction to our local mempool.
+				// Note that we only do this after fetching
+				// the full raw transaction from bitcoind.
+				// We do this so that if that call happens to
+				// initially fail, then we will retry it on the
+				// next interval since it is still not in our
+				// local mempool.
+				b.mempool.add(txHash)
+
+				b.notifyTx(tx.MsgTx())
+			}
+
+		case <-b.quit:
+			return
+		}
+	}
+}
+
+// evictOldTransactions periodically removes any old transactions from our local
+// mempool.
+func (b *bitcoindRPCPollingEvents) evictOldTransactions() {
+	defer b.wg.Done()
+
+	ticker := time.NewTicker(b.cfg.MempoolEvictionAge)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			b.mempool.removeOldTxs(b.cfg.MempoolEvictionAge)
+
+		case <-b.quit:
+			return
+		}
+	}
+}
+
+// mempool represents our view of the mempool and helps to keep track of which
+// mempool transactions we already know about and also what time we saw them.
+type mempool struct {
+	sync.RWMutex
+	txs map[chainhash.Hash]time.Time
+}
+
+// newMempool creates a new mempool object.
+func newMempool() *mempool {
+	return &mempool{
+		txs: make(map[chainhash.Hash]time.Time),
+	}
+}
+
+// clean removes any of the given transactions from the mempool if they are
+// found there.
+func (m *mempool) clean(txs []*wire.MsgTx) {
+	m.Lock()
+	defer m.Unlock()
+
+	for _, tx := range txs {
+		// If the transaction is in our mempool map, we need to delete
+		// it.
+		if _, ok := m.txs[tx.TxHash()]; ok {
+			delete(m.txs, tx.TxHash())
+		}
+	}
+}
+
+// contains returns true if the given transaction hash is already in our
+// mempool.
+func (m *mempool) contains(hash *chainhash.Hash) bool {
```

Comment:
I think we should avoid passing a pointer here since that comes from a range/loop variable. Should probably de-reference it outside of this method (same for `add()`).